test/e2e/rollout/rollout_test.go

```diff
@@ -0,0 +1,313 @@
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, group := range unavailableDevices {
+				Expect(len(group)).To(BeNumerically("<=", 1), "Should have at most 1 unavailable device per group")
+			}
+
+			harness.WaitForBatchCompletion(fleetName, 2, 1*time.Minute)
+
+			By("Verifying all devices are eventually updated")
+			updatedDevices, err := harness.GetUpdatedDevices(fleetName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(updatedDevices)).To(Equal(4), "All devices should be updated eventually")
+		})
+	})
+
+	Context("Success Threshold", func() {
+		It("should pause the rollout if the success threshold is not met", func() {
+			By("creating a fleet with success threshold")
+
+			// Create fleet with 100% success threshold
+			err := harness.CreateTestFleetWithSpec(fleetName, createFleetSpec(fleetName, bsqSuccess, lo.ToPtr(api.Percentage("100%"))))
+			Expect(err).ToNot(HaveOccurred())
+
+			deviceIDs := harness.StartMultipleVMAndEnroll(2)
+
+			labelsList := []map[string]string{
+				{"site": "madrid"},
+				{"site": "paris"},
+			}
+
+			err = harness.SetLabelsForDevicesByIndex(deviceIDs, labelsList, fleetName)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = applicationSpec.FromImageApplicationProviderSpec(applicationConfig)
+			Expect(err).ToNot(HaveOccurred())
+
+			deviceSpec := api.DeviceSpec{
+				Applications: &[]api.ApplicationProviderSpec{applicationSpec},
+			}
+
+			err = harness.CreateOrUpdateTestFleet(fleetName, testFleetSelector, deviceSpec)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Simulating a failure in the first batch")
+			err = SimulateDeviceFailure(harness)
+			Expect(err).ToNot(HaveOccurred())
+
+			time.Sleep(30 * time.Second)
+
+			By("Verifying that the rollout is paused due to unmet success threshold")
+			rolloutStatus, err := harness.GetRolloutStatus(fleetName)
+			Expect(err).ToNot(HaveOccurred())
+			//check rollout status tand reason to be api.ConditionStatusFalse and api.RolloutWaitingReason
+			Expect(rolloutStatus.Status).To(Equal(api.ConditionStatusFalse), "Rollout should be paused when success threshold is not met")
+			Expect(rolloutStatus.Reason).To(Equal(api.RolloutWaitingReason), "Rollout should be paused when success threshold is not met")
+
+			By("Fixing the failed device and verifying the rollout continues")
+			err = FixDeviceFailure(harness)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Wait for rollout to continue
+			harness.WaitForBatchCompletion(fleetName, 2, 1*time.Minute)
+
+			// Verify that all devices are eventually updated
+			updatedDevices, err := harness.GetUpdatedDevices(fleetName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(updatedDevices)).To(Equal(2), "All devices should be updated once threshold is met again")
+		})
+	})
+})
+
+func SimulateDeviceFailure(h *e2e.Harness) error {
+	// Define a command to simulate failure (e.g., stop a critical service or crash the VM)
+	failureCommand := []string{"sudo", "shutdown", "now"} // Example: Shut down the VM
```

Comment:
Instead of  stopping the vm could we systemctl stop flightctl-agent or enable a firewall rule?